documentation/processing.tex

```diff
@@ -26,16 +22,60 @@ \subsection{Overview}
+	
+	\item[Instruction format validators] (see \cref{checker}) are helper objects used by instruction processors. They are provided operands of an instruction and serve to validate their correctness prior to the instruction. 
+	
+\end{description}
+
+Processing manager encapsulates above mentioned objects and determines which processor/provider will be used next (see \cref{fig06:proc_mngr}).
+
+
+\subsection{The main loop of manager}
+\label{lab06:mngr_loop}
+
+Processing manager contains an array of active statement processors and an array of active statement providers. It is in the control of which processor--provider pair currently operates. Moreover, it implements interfaces that swap these objects.
+
+The main processing loop works with the currently operating processor and provider. In the loop body, statement provider provides next statement for statement processor that processes it accordingly. The loop breaks when the last processor finishes work.
 
+When provider ends its statement stream or processor finishes its work, it is replaced with another. The following rules apply:
+
+\begin{enumerate}
+	\item When a processor finishes its work, the next processor is selected from the array.
+	\item When a provider finishes --- before the next provider is selected from the array --- manager checks whether it triggers the termination of the current processor as well (see \emph{terminal condition} in \cref{lab06:sect_proc}). If true, perform rule 1.
+\end{enumerate}
+
+\subsection{Initial state of manager}
+\label{lab06:lib_data}
+
+It is important which statement provider and processor is assigned to the manager as a default. The manager determines this from library data passed by analyzer.
+
+Library data contain a file name and an enumeration indicating a kind of the file that is being parsed --- \emph{processing kind}.
+
+\emph{Ordinary} processing kind states that the file being processed is the main source file (in HLASM called open-code). It is the first file to be processed. With this information, manager initializes all statement providers in respective order and \emph{only} opencode processor. This enumeration is passed when analyzer has owner semantics.
+
+\emph{Copy} and \emph{Macro} processing kinds state that manager will process source code that contains copy or macro definition respectively. Hence, \emph{only} copy definition processor or macro definition processor is initialized. Also, all but macro statement provider are initialized. That is because macro definitions would pollute statement sources (i.e. when parse library provider is called within macro call). Also, no macros will be visited nor needed as a statement source when processing new source code. The library data is passed when analyzer has reference semantics.
```

Comment:
Also, all but macro statement provider are initialized -> Also, all the providers but the macro ...

Reply:
pollute statement sources - tomu nerozumiem

Reply:
pollute davam to prec, je to velky detail